crates/ruff_python_ast/src/source_code/indexer.rs

```diff
@@ -97,6 +111,12 @@ impl Indexer {
         &self.triple_quoted_string_ranges
     }
 
+    /// Return a slice of all ranges that include an f- string. The ranges are sorted by
+    /// [`TextRange::start`] in increasing order. No two ranges are overlapping.
+    pub fn f_string_ranges(&self) -> &[TextRange] {
+        &self.f_string_ranges
+    }
```

Comment:
I would prefer to avoid exposing too much of the assumption on how the ranges is stored and instead provide semantic methods to find an fstring, given a range.